kura/org.eclipse.kura.protocol.can.test/src/main/java/org/eclipse/kura/protocol/can/test/CanSocketTest.java

```diff
@@ -117,6 +101,37 @@ public void updated(Map<String, Object> properties) {
         }
     }
 
+    @Before
+    public void setup() {
+        Assume.assumeFalse(IS_MAC);
+    }
+
+    @Test
+    public void startCanTestThread() {
+        if (this.m_pollThread != null) {
```

Comment:
This code to let the thread finish if there already exists one is as same as in the `deactivate()` method. You can extract this code to a private method.